apps/explorer/src/ui/Search.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { Combobox } from '@headlessui/react';
+import { Search16 } from '@mysten/icons';
+
+import { LoadingSpinner } from './LoadingSpinner';
+import { Text } from './Text';
+
+export type SearchResult = {
+    id: string;
+    label: string;
+    type: string;
+};
+
+export type SearchResults = {
+    label: string;
+    results: SearchResult[];
+};
+
+export interface SearchProps {
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    onSelectResult: (result: SearchResult) => void;
+    placeholder?: string;
+    isLoading: boolean;
+    options?: SearchResults[];
+    value?: SearchResult;
+    queryValue: string;
+}
+
+export interface SearchResultProps {
+    key: string;
+    value: SearchResult;
+    children: React.ReactNode;
+}
+
+function SearchItem({ value, children }: SearchResultProps) {
+    return (
+        <Combobox.Option
+            className="cursor-pointer rounded-md py-1.5 pl-2 ui-active:bg-sui/10 ui-active:shadow-sm"
+            value={value}
+            key={value.id}
+        >
+            <Text variant="body/medium" mono color="steel-darker">
+                {children}
+            </Text>
+        </Combobox.Option>
+    );
+}
+
+export function Search({
+    onChange,
+    onSelectResult,
+    placeholder,
+    options = [],
+    isLoading = false,
+    queryValue,
+    value,
+}: SearchProps) {
+    const hasOptions = options.some((group) => group.results.length > 0);
```

Comment:
Seems like we should just not return if there's no results in the group.

Reply:
we return the search input but not the results UI